plugins/registry.js

```diff
@@ -608,9 +608,9 @@ export default class PluginRegistry {
 
         return {
             id,
-            showRHSPlugin: showRHSPlugin(id),
-            hideRHSPlugin: hideRHSPlugin(id),
-            toggleRHSPlugin: toggleRHSPlugin(id),
+            showRHSPlugin: showRHSPlugin(this.id),
```

Comment:
Apologies for the thrashing, but I'm now better understanding the scope of the changes here -- does this mean a plugin can now register at most one RHS component? Would that be a breaking change for some plugins?

I wonder if we should just close the RHS when a plugin is unloaded, full stop, along with expunging all Redux state associated with a plugin. Then the empty RHS will never occur, albeit at the cost of no longer preserving any state between plugin instances. (Though, is effecting such a change complicated by the pending Redux changes to properly emit the actions when this occurs?) 

Using the force, I imagine that it's safer for plugins (and their registered components) to start from a clean slate than to assume they know how to handle old Redux state from a previous instance. 

Thoughts?

Reply:
Ahh hadn't considered that we allowed registering for multiple plugin components 🤔 as I don't think that's a common practice. Although not impossible. K let me see if it's easy enough to dispatch closing the RHS action.

Reply:
My initial assumption was that plugins couldn't register multiple component, which is why I hadn't gone down the path of closing RHS as I personally think that's a bit destructive (on upgrades).

However seeing that my assumption turned out to be incorrect, the only way to fix this /is/ to close RHS as we have no way of knowing what to replace RHS with post-upgrade.

Updated code to close RHS on plugin upgrade/removal.


Requesting re-review from @marianunez and @cpoile hopefully for the last time 😉. Apologies again for the review thrash.